keps/sig-node/20190129-hugepages.md

```diff
@@ -304,24 +313,36 @@ container cgroup sandbox will be configured with the specified limit.
 The `kubelet` will ensure the `hugetlb` has no usage charged to the pod level
 cgroup sandbox prior to deleting the pod to ensure all resources are reclaimed.
 
-#### Limits and Quota
+##### Limits and Quota
 
 The `ResourceQuota` resource will be extended to support accounting for
 `hugepages-<hugepagesize>` similar to `cpu` and `memory`.  The `LimitRange`
 resource will be extended to define min and max constraints for `hugepages`
 similar to `cpu` and `memory`.
 
-#### Scheduler changes
+##### Scheduler changes
 
 The scheduler will need to ensure any huge page request defined in the pod spec
 can be fulfilled by a candidate node.
 
-#### cAdvisor changes
+#### cAdvisor changes(Phase 1)
 
 cAdvisor will need to be modified to return the number of pre-allocated huge
 pages per page size on the node.  It will be used to determine capacity and
 calculate allocatable values on the node.
 
+### Phase 2
+
+#### Support container isolation of huge pages
+
+Container isolation of huge pages should be supported to avoid competition between containers to consume huge pages. Currently, `kubelet` directly sets the aggregated huge pages limits on pod's cgroup of hugetlb subsystem. The limits should be set on a container's cgroup. To achieve this, the `LinuxContainerResources` message which is used to update the container resource should be updated to specify huge page limits per container. `containerd` also should be updated to handle huge page.
+
+#### cAdviser changes(Phase 2)
+To support NUMA, `cAdviser` should discover and store pre-allocated huge pages per NUMA node. The `v3` version of `MachineInfo` will be introduced.
+
+#### Enhance Node Allocatable feature to reserve huge pages for system
+Some system services like `OVS-DPDK` comsume huge pages per NUMA node, to determine the allocatalbe number of huge pages in `kubelet`, `Node Allocatable feature` should support to reserve huge pages per NUMA node.
```

Comment:
Not sure if I understand this correct; do you mean that huge pages per NUMA node should be node resources in k8s in the same way as huge pages (and cpu, memory, and empherial storage) is today? Or do you mean that the data should be stored internally, so that it can be utilized by the topology manager?

Reply:
At this point, the data which represents pre-allocated hugepages per NUMA node will be additional information for future usage(Memory Manager or maybe Topology Manager).
So, I mean the data should be stored internally.

And I believe that the node resources that you mentioned are tightly coupled with node scheduler.